src/test/shared/featureController.test.ts

```diff
@@ -0,0 +1,65 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import { FeatureController } from '../../shared/featureController'
+import { TestSettingsConfiguration } from '../utilities/testSettingsConfiguration'
+
+describe('FeatureController', async () => {
+
+    describe('isFeatureActive', async () => {
```

Comment:
No tests exercise the codepath where `overrideKeys` is not passed in. 

Would it be possible to iterate all of the enum keys, and test that isFeatureActive returns true for each one dynamically (search for `samCliVersionTestScenarios` for an example)? This would auto-validate as features are added/removed to the enum.

If not, consider adding a comment in the enum to add an explicit test for any feature added to the enum.

Reply:
Would this work:
```
        for (const featureFlag of Object.keys(ActiveFeatureKeys)) {
            it(`returns true for ${featureFlag}`, async () => {
                // simulate settings.json
                const config = new TestSettingsConfiguration()
                await config.writeSetting('experimentalFeatureFlags', [featureFlag])

                // simulate active feature flags
                const features = new FeatureController(config)

                assert.ok(
                    features.isFeatureActive(featureFlag),
                    `Expected ${featureFlag} from ActiveFeatureKeys to be active`
                )
            })
        }

```

Reply:
Looks like it should. It's not a direct test of how a user would use it (since the test doesn't pick up the enum itself), but that's also partially why we have `isFeatureActive` use a string instead of an enum directly. 

Which do you prefer? Testing to ensure the enums reflect features, or testing to make sure that `isFeatureActive(ActiveFeatureKeys.${feature})` works?

Reply:
All of the tests in this file rely on `overrideKeys`, which means we don't test the behavior that will be used by the toolkit.  The purpose isn't to test whether the enums reflect features, rather to verify that the code path that relies on the enum directly works as expected. 

It's a good point that testing each enum value is not necessary. Instead, if the enum is not empty, we could perform the test with the only first entry, and that would be sufficient. The test would be renamed to something like "returns true when not using overrideKeys"